_rules/required-form-fields-not-completed-334972.md

```diff
@@ -0,0 +1,238 @@
+---
+id: 334972
+name: Error message identifies required fields left unfilled
+rule_type: atomic
+description: |
+  This rule checks that text error messages identify required form fields that were not completed.
+accessibility_requirements:
+  wcag20:3.3.1: # Error Identification (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G83: # Providing text descriptions to identify required fields that were not completed
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree # (e.g. HTTP Messages, DOM Tree, CSS Styling, Accessibility Tree, Language, etc.,)
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
+
+The rule applies to each [required][] HTML element that has one of the following [semantic roles][semantic role]: `checkbox`, `combobox`, `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`.
+
+**Note**: The list of applicable [semantic roles][semantic role] is derived by taking all the [ARIA 1.1][] roles that:
+
+- inherit from the [abstract][] `input` or `select` role, and
+- do not have a [required context][] role that itself inherits from one of those roles.
+
+## Expectation 1
+
+For each test target where the lack of user input caused the appearance of an [error message][], the error message is [visible][] and [included in the accessibility tree][].
```

Comment:
"lack of user input" is ambiguous. For example, someone may interpret this to mean that the user let a session expire because they went away from their machine for too long.  Failing that is not the intent of this success criterion, but the way this is written it could be misunderstood to mean this.

I also don't think the lack of input causes anything. Submitting a form causes the error. I think this could be phrased a little clearer.

Reply:
I think you're backtracking a little here. We discussed this previously and because of the difficulty in objectively identifying all situations that could cause an error message the expectation focused on an error message appearing, not on what caused its appearance (e.g. submitting a form). That makes sense because the SC is concerned with the contents of the error message, not on the mechanisms (client- or served-based) that create the error message. Even if the user lets the session expire and because of that an error message is displayed, the error message should still identify the fields in an error state.

Reply:
Still, I agree that we should leave causation out of the expectation and I'll replace "caused" with "resulted in". (I'll do the same for the other rule in this PR)